File: 프로그래머스/lv1/12935. 제일 작은 수 제거하기/제일 작은 수 제거하기.py
def solution(arr):
    length = len(arr)
    answer = []
    if length == 1:
        answer = [-1]
    else:
        min_num = min(arr)      # 제일 작은 수 확인
        flag = True            # 아직 제일 작은 수 제외 못함
        for n in range(length):
            if arr[n] == min_num and flag:       # 제일 작은 수 & 제거도 하기 전이라면 추가하지 않음
                continue
            answer += [arr[n]]
                
    return answer

File: 프로그래머스/1/12935. 제일 작은 수 제거하기/제일 작은 수 제거하기.java
import java.util.*;

class Solution {
    public int[] solution(int[] arr) {
        if (arr.length == 1) {      // 숫자가 한 개만 있으면 [-1] 반환
            int[] answer = {-1};
            return answer;
        }
        
        int[] answer = new int[arr.length - 1];
        int min = Arrays.stream(arr).min().getAsInt();      // 배열의 최솟값
        int i = 0;
        
        for (int num : arr) {
            if (num == min) {           // 최소 숫자는 제외
                continue;
            }
            answer[i] = num;
            i++;
        }
        
        return answer;
    }
}